src/kvstore/raftex/RaftPart.cpp

```diff
@@ -1300,14 +1299,20 @@ void RaftPart::processAskForVoteRequest(
     lastMsgRecvDur_.reset();
     weight_ = 1;
 
+    if (req.get_last_log_term() < wal_->lastLogTerm()) {
+        resp.set_error_code(cpp2::ErrorCode::E_TERM_OUT_OF_DATE);
+        return;
+    }
+
+    if (req.get_last_log_id() < wal_->lastLogId()) {
+        resp.set_error_code(cpp2::ErrorCode::E_LOG_STALE);
+        return;
+    }
+    resp.set_error_code(cpp2::ErrorCode::SUCCEEDED);
+
     // If the partition used to be a leader, need to fire the callback
     if (oldRole == Role::LEADER) {
         LOG(INFO) << idStr_ << "Was a leader, need to do some clean-up";
-        if (wal_->lastLogId() > lastLogId_) {
-            LOG(INFO) << idStr_ << "There is one log " << wal_->lastLogId()
-                      << " i did not commit when i was leader, rollback to " << lastLogId_;
-            wal_->rollbackToLog(lastLogId_);
```

Comment:
I agree with that we should not rollback when a host was leader before, at least according to standard raft, new leader would correct these log for old leader. 

But there is a problem that if we do rollback here, say ABC, A was leader of Term 1 and has one more log than B and C(e.g. it has not been sent out), and then B is elected as Term 2. If A didn't rollback the extra log, A will never accept any log from B, because we restrict the wal log id must be consecutive.

Reply:
So here we compare term and id as a whole.